### PR TITLE
Move compile in HttpApp to separate module

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/HttpCompile.scala [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import io.netty.handler.codec.http._
+import zhttp.experiment.ContentDecoder
+import zhttp.http.HttpApp.InvalidMessage
+import zhttp.http._
+import zio.stream.ZStream
+import zio.{Chunk, Promise, UIO, ZIO}
+
+import java.net.{InetAddress, InetSocketAddress}
+
+final case class HttpCompile[R, E] private[zhttp] (app: HttpApp[R, E], zExec: HttpRuntime[R])
```
Comment (ShrutiVerma97) [inline zio-http/src/main/scala/zhttp/service/HttpCompile.scala]: HttpCompile should be renamed as Handler

File: zio-http/src/main/scala/zhttp/service/HttpCompile.scala [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+object HttpCompile {
+  def apply[R, E](app: HttpApp[R, E], zExec: HttpRuntime[R])(implicit
```
Comment (ShrutiVerma97) [inline zio-http/src/main/scala/zhttp/service/HttpCompile.scala]: remove apply method, case class should extend ChannelInboundHandlerAdapter directly

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 3]
```diff
@@ -23,19 +17,6 @@ case class HttpApp[-R, +E](asHttp: Http[R, E, Request, Response[R, E]]) { self =
 
   def +++[R1 <: R, E1 >: E](other: HttpApp[R1, E1]): HttpApp[R1, E1] = self defaultWith other
 
-  /**
-   * Converts a failing Http app into a non-failing one by handling the failure and converting it to a result if
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala:27]: why is this deleted?

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 4]
```diff
@@ -384,10 +140,4 @@ object HttpApp {
   def fromFunction[R, E, B](f: Request => HttpApp[R, E]): HttpApp[R, E] =
     HttpApp(Http.fromFunction[Request](f(_).asHttp).flatten)
 
-  /**
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala:387]: Also this? why was it deleted
